CONTRIBUTING.md

```diff
@@ -150,3 +150,49 @@ tools this can increase the effort required to review a submission. Be careful w
 The automatic checking system will run all our unit tests and integration tests across several systems. However, this
 process will take time. Try to run the unit tests locally during development before submitting a PR and use the
 automatic checks as a safety net.
+
+The tests can be run using the following commands:
+
+```bash
+# Go into your FPP directory
+cd MY_FPP_DIRECTORY
+
+# Run CI tests on fprime
+./ci/tests/Framework.bash
+
+# Run CI tests on the reference application
+./ci/tests/Ref.bash
+
+# Run the static analyzer
+# Purge all directory
+fprime-util purge -f
+# Generate the build files for clang-tidy. Make sure clang-tidy is installed.
+fprime-util generate -DCMAKE_CXX_CLANG_TIDY="clang-tidy-12;--config-file=$PWD/release.clang-tidy"
+# Build fprime with the static analyzer
+fprime-util build -j4
+```
+
+## Development with modified FPP version
+
+In case FPP needs to be locally changed, FPP first needs to be installed:
+
+```bash
+# Go into your FPP directory
+cd MY_FPP_DIRECTORY
+# Install FPP. Make sure to have sbt installed.
+./compiler/install
+# Copy
+cp ./compiler/bin/* /usr/local/bin/ -r
```

Comment:
I personally do
```
export PATH=<path/to/fpp>/compiler/bin:$PATH
```
Do you think it might be better to recommend that in the guide? Forgetting to uninstall could lead to very hard to debug issues

Reply:
I definitely think that there is a better way than mine.

I feel like, ideally, there should be a `setup.py` that we could use to install the fpp packages with `pip install .` after the command `compile/install` has been called. I know the `setup.py` file existed in the past (https://github.com/nasa/fpp/blob/v2.0.1/setup.py), but I'm not sure what was the reason why it was removed.

As you suggested, the `export PATH` way is already better than what I am currently doing.

I have to say that I am not using any virtual environment in my development routine right now, since I am having a dedicated docker image for fprime. However, what would be the command to add it to the virtual environment? 
Maybe both approaches can be suggested.

Reply:
Do the instructions here work: https://github.com/nasa/fpp/blob/main/compiler/README.adoc?